### PR TITLE
Remove tempfile file generator

### PR DESCRIPTION
**Pull request by tonynajjar**
Fixes https://github.com/osrf/rocker/issues/187

### CODE REVIEW

File: src/rocker/nvidia_extension.py [hunk 1]
```diff
@@ -42,18 +40,18 @@ def get_name():
     def __init__(self):
         self.name = X11.get_name()
         self._env_subs = None
-        self._xauth = tempfile.NamedTemporaryFile(prefix='.docker', suffix='.xauth')
```
Comment (tonynajjar) [inline src/rocker/nvidia_extension.py:45]: There is probably a reason behind this?

Comment (tfoote) [reply src/rocker/nvidia_extension.py:45]: This was made unique to keep multiple containers from interfering with each other.

The tempfile is the safe way to make a file like this to ensure that there's no collisions etc.

What I would recommend instead is to extend the nocleanup logic to make the tempfile persist. 

An extension would be to also think about how to have a second step which would clean up any dangling resources like this for a container previously passed nocleanup

**Comment by tfoote** [comment]
Could you take a look at #189 as an alternative?

**Comment by tonynajjar** [comment]
Hey, sorry for the late reply, I'll test it out by the end of the week

**Comment by tonynajjar** [comment]
Works like a charm, closing this PR

**Pull request closed without merging**